jpa/subsystem/src/main/java/org/jboss/as/jpa/transaction/TransactionUtil.java

```diff
@@ -45,19 +53,29 @@
 
     private static volatile TransactionSynchronizationRegistry transactionSynchronizationRegistry;
     private static volatile TransactionManager transactionManager;
+    private static volatile TransactionListenerRegistry transactionListenerRegistry;
```

Comment:
We need to reference the statics from non-service code (e.g. container managed persistence context).  We sort of guard the statics with the JPAService but that could be improved (the JPAService injector.uninject methods should probably null the statics out).  Beyond that, any other suggestions for improvement here?